utils/utils.go

```diff
@@ -0,0 +1,57 @@
+package utils
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/go-openapi/runtime"
+)
+
+// ClientConfig contains configuration passed to the individual API Clients.
+type ClientConfig struct {
+	// NumRetries contains the number of attempted retries
+	NumRetries int
+	// RetryTimeout says how long to wait before retrying a request
+	RetryTimeout time.Duration
+	// RetryStatusCodes contains the list of status codes to retry, use "x" as a wildcard for a single digit (default: [429, 5xx])
+	RetryStatusCodes []string
+}
+
+func GetTimeout(retryTimeout time.Duration) time.Duration {
+	if retryTimeout == 0 {
+		return time.Second * 5
+	}
+	return retryTimeout
+}
+
+// MatchRetryCode checks if the status code matches any of the configured retry status codes.
+func MatchRetryCode(err error, retryCodes []string) (bool, error) {
+	if len(retryCodes) == 0 {
+		retryCodes = []string{"429", "5xx"}
```

Comment:
@zserge @papagian @julienduchesne: The manual client hardcoded retries for `429` and `5xx` API errors, even if the user did not set this in the Client config. Is this default something that we would like to continue to support? Please share your views & different perspectives on this :)

Reply:
I don't really know and have a strong opinion: either keep it as is or change it and make sure we document it properly.

However, I wonder why `retryCodes` is not an array of integers instead of strings. In that case the above check would become as simple as:
`respCode == StatusTooManyRequests || respCode > http.StatusInternalServerError`

Reply:
They are strings so that they can be set as `5xx`

Reply:
Agree it would useful to allow configuration of the status codes. Do we expect the manual client to be reused in future or do we expect this generated client to be the most commonly used? I think it would be better to keep them consistent and also I'm in favour of having sensible defaults, so keeping [429, 5xx] codes. +1 with @papagian to make sure we document it properly.

Reply:
> Do we expect the manual client to be reused in future or do we expect this generated client to be the most commonly used?

The goal is to [deprecate the manual client](https://github.com/grafana/grafana/issues/71996) and use this generated client. We will be ready to do that once we have the automation in place and some services (such as `Foder`) that is fully working and tested E2E. 

Sounds good, will add more documentation around the possible configurations and any defaults :)